analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.LearnConstraint.cs

```diff
@@ -433,6 +433,59 @@ public void Branching_LearnsObjectConstraint_TypePattern_Negated(string expressi
                 .And.ContainSingle(x => x != null && x.HasConstraint(ObjectConstraint.NotNull));
         }
 
+        [DataTestMethod]
+        [DataRow("arg is { }")]
+        [DataRow("arg is not not { }")]
+        [DataRow("!!(arg is { })")]
+        public void Branching_LearnsObjectConstraint_RecursivePattern_ElseIsNull(string expression)
+        {
+            var validator = CreateIfElseEndValidatorCS(expression, OperationKind.RecursivePattern);
+            validator.ValidateTag("If", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.ValidateTag("Else", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
```

Comment:
Add tests were arg is a value type. There shouldn't be constraint here (there shouldn't be a constraint in if either). Add also tests with the other type kinds (e.g. unconstraint open generic).

Reply:
Hint: This is relevant for other patterns, like #6059 too.